src/rules/selector-type-case/__tests__/index.js

```diff
@@ -0,0 +1,76 @@
+import {
+  ruleTester,
+} from "../../../testUtils"
+import rule, { ruleName, messages } from ".."
+
+const testRule = ruleTester(rule, ruleName)
+
+testRule("lower", tr => {
+    
+  tr.ok("a { display: inline-block; }")
+  tr.ok("div { display: inline-block; }")
+  tr.ok("body { display: inline-block; }")
+  tr.ok("li { display: inline-block; }")
+
+  tr.ok("a::before { content: \"some content\"; }")
+  tr.ok("div::before { display: inline-block; }")
+  tr.ok("body::before { display: inline-block; }")
+  tr.ok("li::before { display: inline-block; }")
+
+  tr.ok("&a { display: inline-block; }")
+  tr.ok("&li { display: inline-block; }")
+
+  tr.ok(".foo { display: inline-block; }")
+  tr.ok("#bar { display: inline-block; }")
+
+  tr.ok(".FOO { display: inline-block; }")
+  tr.ok("#BAR { display: inline-block; }")
+
+  tr.notOk("A { color: #Ababa; }", {
+    message: messages.expected("A", "a"),
+  })
+  tr.notOk("DIV { display: inline-block; }", {
```

Comment:
One warning simple type selector should be enough. Instead, can you create equivalent warnings tests for the compound selector, complex selector, single nesting selector and a selector list example I gave above?